File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.py
def solution(n):
    battery = 0
    # 이동 후 홀 짝 기준으로 이동 방법 변경
    while n:
        if (n % 2) == 0:
            n = n // 2
        else:
            battery += 1
            n = (n - 1) // 2

    return battery

File: 프로그래머스/2/12980. 점프와 순간 이동/점프와 순간 이동.java
import java.util.*;

public class Solution {
    public int solution(int n) {
        // 한 번에 k 칸 앞으로 점프 -> k만큼의 건전지 소모
        // 현재까지 온 거리 * 2 순간이동 가능 -> 건전지 소모 없음
        String binaryNum = Integer.toBinaryString(n);
        
        int result = Integer.bitCount(n);
        
        return result;
    }
}